pkg/controller/worker/machines.go

```diff
@@ -268,3 +274,35 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 
 	return nil
 }
+
+func (w *workerDelegate) enableAcceleratedNetworking(machineTypeName string, machineImageConfig *extensionsv1alpha1.MachineImage) bool {
+	if machineImageConfig == nil {
+		return false
+	}
+
+	// Check if used operating system and version are supporting accelerated networking.
+	var osSupport bool
+	for _, image := range w.cloudProfileConfig.MachineImages {
```

Comment:
There is already a loop over the `MachineImages` here: https://github.com/gardener/gardener-extension-provider-azure/pull/65/files#diff-30e669b862b2c1062573be5e967954c3R128

What about enhancing this function to not only return `urn` and `id` but also this information about the `acceleratedNetworking`? You could pass this as a parameter to this function and then you only need to loop over the machine types below.

Reply:
Sounds reasonable for me also as we could also define accelerated networking capabilities in the `machineImage` section of the worker status.